### ISSUE TITLE
Forbid initializing 2 or more variables on one line

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1901]
# Rule request

## Thesis

```python
# bad
ab, cd = [], []

# good
ab = []
cd = []
```

## Reasoning

Decrease unnecessary code line complexity.

**Comment by sobolevn** [comment]
One more thing, unpacking is a good thing and should be allowed: `a, b = some_tuple()`
The thing about this example is that `[], []` does not look like a real tuple. But, for example: `x, y = (1, 2)` looks pretty readable to me.

So, we need to find a balance between an obviously good idea - one assignment per line - and some real-world usages, when this rule is not applicable.

**Comment by orsinium** [comment]
> One more thing, unpacking is a good thing and should be allowed: a, b = some_tuple()

Exactly 👍🏿 

> But, for example: x, y = (1, 2) looks pretty readable to me.

I don't think it will drop readability if it is on separate lines. Moreover, now I work on a codebase that has plenty of it and I always miss where the initialization of the second variable (`y` in your example) happens. So, I don't have cases in mind when one-line initialization is useful.

Ok, there is one case: swapping variables (`x, y = y, x`). However, I can't recall any case when I actually used it. Non-recursive GCD, maybe.

**Comment by sobolevn** [comment]
> swapping variables

Yes, sometimes I use this when doing non-recursive things like tree traversal.

**Comment by f1sherFM** [comment]
@sobolevn Can i take this?

**Comment by Khabib73** [comment]
I’m just a passerby, but I’ve been following this discussion :) Given the maintainer’s feedback about tuple unpacking being idiomatic, this proposal seems too broad and hard to narrow down without false positives.

Since there hasn’t been further activity and the idea appears to be abandoned, maybe it’s best to close this issue for now to reduce noise?

**Comment by sobolevn** [comment]
Let's keep it open, maybe one day someone will get back to it :)

**Comment by dreamflasher** [comment]
Also there is already an open PR for this that waits for review?
